### PR TITLE
Pass the appname flag as part of user agent to Go storage library

### PR DESCRIPTION
**Pull request by Tulsishah**
Passing AppName flag as part of user agent to Go Storage library. Checking Environment variable for DLVM and DLC and adding that into UserAgent.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -139,12 +139,26 @@ func getConnWithRetry(flags *flagStorage) (c *gcsx.Connection, err error) {
 ////////////////////////////////////////////////////////////////////////
 
 // Mount the file system according to arguments in the supplied context.
+
+func getUserAgent(AppName string) string {
+	var userAgent string
+
+	if os.Getenv("GCSFUSE_METADATA_IMAGE_TYPE") != "" {
```
Comment (sethiay) [inline main.go:146]: Do you also think we can simplify it by setting user agent like:
userAgent = strings.TrimSpace(fmt.Sprintf("gcsfuse/%s %s %s", getVersion(), appName, os.Getenv("GCSFUSE_METADATA_IMAGE_TYPE")))

File: main.go [hunk 2]
```diff
@@ -139,12 +139,26 @@ func getConnWithRetry(flags *flagStorage) (c *gcsx.Connection, err error) {
 ////////////////////////////////////////////////////////////////////////
 
 // Mount the file system according to arguments in the supplied context.
+
+func getUserAgent(AppName string) string {
```
Comment (sethiay) [inline main.go:143]: AppName should be in camelCase ? i.e. appName

File: main_test.go [hunk 3]
```diff
@@ -25,3 +26,34 @@ func (t *MainTest) TestCreateStorageHandleEnableStorageClientLibraryIsTrue() {
...
+func (t *MainTest) TestGetUserAgentWhereEnvironmentVariableEqualDLVM() {
+	os.Setenv("GCSFUSE_METADATA_IMAGE_TYPE", "DLVM")
+	defer os.Unsetenv("GCSFUSE_METADATA_IMAGE_TYPE")
+
+	userAgent := getUserAgent("AppName")
+
+	ExpectEq("gcsfuse/unknown (Go version go1.20-pre3 cl/474093167 +a813be86df) AppName DLVM", userAgent)
+}
+
+func (t *MainTest) TestGetUserAgentEqualNil() {
```
Comment (sethiay) [inline main_test.go:55]: UserAgent is not Nil but the environment variable is not set. So, can we change the name of function to reflect this case ?

File: main.go [hunk 4]
```diff
@@ -153,6 +167,7 @@ func createStorageHandle(flags *flagStorage) (storageHandle storage.StorageHandl
 		HttpClientTimeout:   flags.HttpClientTimeout,
 		MaxRetryDuration:    flags.MaxRetryDuration,
 		RetryMultiplier:     flags.RetryMultiplier,
+		UserAgent:           getUserAgent(flags.AppName),
```
Comment (sethiay) [inline main.go:170]: Your current changes takes care of user agent in Storage client library. We also need to take care when used with jacobsa/gcloud: https://github.com/GoogleCloudPlatform/gcsfuse/blob/5a4355196a0a551505f1175195eafece577a82e4/main.go#L98

File: internal/storage/storage_handle_test.go [hunk 5]
```diff
@@ -34,6 +34,7 @@ func getDefaultStorageClientConfig() (clientConfig StorageClientConfig) {
 		HttpClientTimeout:   800 * time.Millisecond,
 		MaxRetryDuration:    30 * time.Second,
 		RetryMultiplier:     2,
+		UserAgent:           "AppName",
```
Comment (sethiay) [inline internal/storage/storage_handle_test.go:37]: Can we set some close to real-world UserAgent ?

File: internal/storage/storage_handle_test.go [hunk 6]
```diff
@@ -100,3 +101,10 @@ func (t *StorageHandleTest) TestNewStorageHandleWithZeroMaxConnsPerHost() {
 
 	t.invokeAndVerifyStorageHandle(sc)
 }
+
+func (t *StorageHandleTest) TestNewStorageHandleWithDifferentUserAgent() {
+	sc := getDefaultStorageClientConfig()
+	sc.UserAgent = "UserAgent"
```
Comment (sethiay) [inline internal/storage/storage_handle_test.go:107]: Same: Can we set some close to real-world UserAgent ?

**Review by sethiay — CHANGES_REQUESTED** [review]

File: main.go (hunk 1 above)
Comment (Tulsishah) [reply main.go:147]: fixed

File: main_test.go (hunk 3 above)
Comment (Tulsishah) [reply main_test.go]: fixed

File: main.go (hunk 4 above)
Comment (Tulsishah) [reply main.go:171]: fixed

File: internal/storage/storage_handle_test.go (hunk 5 above)
Comment (Tulsishah) [reply internal/storage/storage_handle_test.go]: fixed

File: main.go (hunk 2 above)
Comment (Tulsishah) [reply main.go]: fixed

File: internal/storage/storage_handle_test.go (hunk 6 above)
Comment (Tulsishah) [reply internal/storage/storage_handle_test.go]: fixed

**Pull request closed without merging**